samples/matter/common/src/watchdog/watchdog.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "watchdog/watchdog.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace Nrf;
+
+namespace
+{
+const struct device *const wdt = DEVICE_DT_GET(DT_ALIAS(watchdog0));
+}
+
+bool Watchdog::sEnabled = false;
+
+Watchdog::Watchdog(uint32_t minWindowTime, uint32_t maxWindowTime, uint32_t feedingInterval = 0,
```

Comment:
Yes, the Watchdog module has only several channels to assign, so we can create Watchdog objects as far as a channel is available. I will add an assert there.

Reply:
Ok, I checked that it is handled by the `wdt_install_timeout` function. If there are no more channels available the function returns `-ENOMEM` and it will be blocked by the assert below. I will add a simple log there for developers for this error code.